src/extension/ai/ghost.ts

```diff
@@ -347,13 +342,16 @@ export class CellChangeEventGenerator {
       return
     }
 
-    this.streamCreator.handleEvent(
-      new stream.CellChangeEvent(
-        notebook.uri.toString(),
-        matchedCell.index,
-        StreamGenerateRequest_Trigger.CELL_TEXT_CHANGE,
-      ),
-    )
+    // N.B. handlEvent is aysnc. So we need to use "then" to make sure the event gets processed
+    this.streamCreator
```

Comment:
@sourishkrout Is this the right way to call an async function from a non async function?
Since the return type is null we don't need to await the async function.
However, I think in the past you mentioned that if you invoke an async function but don't do anything with its return value it might not get scheduled. 
Does using "then(()=> {})" solve this problem?

Reply:
It does not solve the problem. At JS runtime, there isn't such a thing as an "async function". What makes it async is returning a `Promise<>` type. Async/await uses a generator under the hood to pause execution to unravel promises; however, semantically, there's no difference to `.then(...)`. It just makes you write async code that looks much more like sync code for readability.

That being said, if a promise type ("a future") is not being then'd or awaited upstream, you wind up with the same exact problem no matter how the promise is "returned."

The only reason this likely appears to work is that promises start running immediately (as opposed to when you await/then them), and VS Code is a long-running process, so the scopes live long enough to allow the promises to complete. I'm making assumptions here because I haven't inspected the upstream code in detail.

As a golang programmer, this is the same as running three `go func()` inside a function but not using a WaitGroup to synchronize them. Similarly, this might work if the "main thread" runs long enough for all three to complete and no downstream processing requires their completion.

In other words, my suggestion to use `async` was only intended to be cosmetic. If there is a problem with async execution, both `then` and `await` will be prone to it.

Reply:
> The only reason this likely appears to work is that promises start running immediately (as opposed to when you await/then them), and VS Code is a long-running process, so the scopes live long enough to allow the promises to complete

That seems fine to me. Concretely, it seems fine to treat it as a fire and forget and assume
1) vscode runs long enough for them to complete 
or
2) If vscode shuts down early enough then the request might not have completed.

Here's the problem I'm trying to solve. handleEvent is an async function being called from a non async function handleOnDidChangeNotebookCell. handleOnDidChangeNotebookCell is the listener for events
https://github.com/stateful/vscode-runme/blob/165da756be8ef14ee604c2b9afb352f26ca2841e/src/extension/ai/manager.ts#L60

I originally thought I couldn't declare handleEvent as async and include an await function because I thought the listener couldn't be an async function. However, it looks like if I declare handleOnDidChangeNotebookCell to be async it still works fine; so I could update it to be async and then include an await function.

However, my suposition is that  if the listener (handleOnDidChangeNotebookCell) is async its returning a promise that is not being awaited on.  So we still wind up with a Promise that we aren't awaiting its just happening in a different part of the code. 

Fundamentally, I think this is alright. The listener (handleOnDidChangeNotebookCell) is firing off a request to generate a completion and the response is handled asynchronously. If vscode exits before that async function can get scheduled and finish processing than we are dropping the completion generation logic on the floor rather than doing some graceful handling/shutdown. But dropping it on the floor seems fine; I'm not sure what graceful handling would actually look like.

Reply:
So I changed handleOnDidChangeNotebookCell and added some awaits. I made this change because I agree with your earlier comment that using await is cleaner.

@sourishkrout so I think this is good to go but let me know if you think otherwise.